app/controllers/api/v1/targets_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Api
+  module V1
+    class TargetsController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_target, only: [:show]
```

Comment:
I would suggest using `helper_method :target` here instead, and change the private method `set_target` to `target`. By doing this, when you call `target` from a view, it goes to this method.